src/panels/lovelace/editor/config-elements/hui-map-card-editor.ts

```diff
@@ -213,6 +240,25 @@ export class HuiMapCardEditor extends LitElement implements LovelaceCardEditor {
     fireEvent(this, "config-changed", { config: ev.detail.value });
   }
 
+  protected willUpdate() {
+    if (this.hass && !this._possibleGeoSources) {
+      const sources: Record<string, string> = {};
+      Object.entries(this.hass.states).forEach(([entity_id, stateObj]) => {
+        const domain = computeDomain(entity_id);
+        if (domain === "geo_location" && stateObj.attributes.source) {
+          sources[stateObj.attributes.source] = stateObj.attributes.attribution;
+        }
+      });
+
+      this._possibleGeoSources = Object.entries(sources).map(
+        ([source, attribution]) => ({
+          value: source,
+          label: attribution || source,
+        })
+      );
```

Comment:
Why use 2 loops to create an object and then an array? Is this for deduplication of sources?

Reply:
Yes was just trying to use an Object to deduplicate the keys, while saving the attribution to go with each key. 

Second loop puts this into Array format as required by the selector. 

It's only executed once.